fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/RegexpReplace.java

```diff
@@ -52,13 +59,44 @@ public RegexpReplace(Expression arg0, Expression arg1, Expression arg2) {
         super("regexp_replace", arg0, arg1, arg2);
     }
 
+    /**
+     * constructor with 4 arguments.
+     */
+    public RegexpReplace(Expression arg0, Expression arg1, Expression arg2, Expression arg3) {
+        super("regexp_replace", arg0, arg1, arg2, arg3);
+    }
+
     /**
      * withChildren.
      */
     @Override
     public RegexpReplace withChildren(List<Expression> children) {
-        Preconditions.checkArgument(children.size() == 3);
-        return new RegexpReplace(children.get(0), children.get(1), children.get(2));
+        Preconditions.checkArgument(children.size() == 3 || children.size() == 4,
+                "RegexpReplace should have 3 or 4 children, but got: " + children.size());
+        if (children.size() == 3) {
+            return new RegexpReplace(children.get(0), children.get(1), children.get(2));
+        } else {
+            return new RegexpReplace(children.get(0), children.get(1), children.get(2), children.get(3));
+        }
+    }
+
+    @Override
+    public void checkLegalityBeforeTypeCoercion() {
+        if (children().size() == 3) {
+            return;
+        }
+        if (children().size() == 4) {
+            Expression value = child(3);
+            DataType type = value.getDataType();
+            if (!type.isStringLikeType()) {
+                throw new AnalysisException(
+                        "The fourth param of regexp_replace must be a string type: " + this.toSql());
+            }
+            if (!value.isConstant()) {
```

Comment:
in planner, `isConstant` means there is not slot in the expr tree. so a function expr could return true when eval `isConstant`. if the last parameter could be a function, why not must be constant?

Reply:
have changed, and I only want the fourth param is string literal